src/lib/redux/siteDataSlice.ts

```diff
@@ -69,9 +92,15 @@ const siteDataSlice = createSlice({
         ]
         }
       }
-      }
+    },
+    editCustomer(state, action) {
+      const stateCopy = JSON.parse(JSON.stringify(state));
+      const index = stateCopy.currentSite.customers.findIndex((e: CustomerRecord) => e.id === action.payload.id);
+      stateCopy.currentSite.customers[index] = action.payload;
```

Comment:
If the customer we are editing is the same as the current customer (which might always be the case?), we need to update the `currentCustomer`'s object so that it also reflects the changes.

Reply:
Maybe the ^ should happen in `editCustomerInRedux`? Do not know, whatever you think is cleaner